pkg/client/client.go

```diff
@@ -467,6 +447,66 @@ func (o *OptimizelyClient) GetAllFeatureVariablesWithDecision(featureKey string,
 	return enabled, variableMap, errs.ErrorOrNil()
 }
 
+// GetDetailedFeatureDecisionUnsafe triggers an impression event and returns all the variables
+// for a given feature along with the experiment key, variation key and the enabled state.
+// Usage of this method is unsafe and not recommended since it can be removed in any of the next releases.
+func (o *OptimizelyClient) GetDetailedFeatureDecisionUnsafe(featureKey string, userContext entities.UserContext, disableTracking bool) (decisionInfo decision.UnsafeFeatureDecisionInfo, err error) {
+
+	decisionInfo = decision.UnsafeFeatureDecisionInfo{}
+	decisionInfo.VariableMap = make(map[string]interface{})
+	decisionContext, featureDecision, err := o.getFeatureDecision(featureKey, "", userContext)
+	if err != nil {
+		o.logger.Error("Optimizely SDK tracking error", err)
+		return decisionInfo, err
+	}
+
+	if featureDecision.Variation != nil {
+		decisionInfo.Enabled = featureDecision.Variation.FeatureEnabled
+		decisionInfo.VariationKey = featureDecision.Variation.Key
+		decisionInfo.ExperimentKey = featureDecision.Experiment.Key
```

Comment:
I think we should only assign these when `featureDecision.Source == decision.FeatureTest`. Keys from a rollout experiment should not be exposed.